packages/typespec-azure-resource-manager/lib/private.models.tsp

```diff
@@ -0,0 +1,10 @@
+namespace Azure.ResourceManager.Private;
+
+/**
+ * Describes the shape of strict constraint model to type check against `Resource` template
+ *  parameter of ARM operation templates & base models.
+ */
+model ArmResourceTypeConstraint {
```

Comment:
Per the issue this morning, this behavior will not work after we fix this issue:https://github.com/microsoft/typespec/issues/3267

So I think we need to go with one of the other options (a decorator, for example, would allow us to make this an error)